### ISSUE TITLE
Incomplete file-type handling

### ISSUE THREAD

**Issue by mcint** [sharkdp/fd#511]
I was looking for sockets, but noticed some other file types are missing as well.

From `find`:
```
     -type t
             True if the file is of the specified type.  Possible file types are as follows:

             b       block special
             c       character special
             d       directory
             f       regular file
             l       symbolic link
             p       FIFO
             s       socket
```
equivalently, via the Open Group [sys/stat.h](https://pubs.opengroup.org/onlinepubs/009604499/basedefs/sys/stat.h.html) linked from Wikipedia
```
S_IFMT  Type of file.
  S_IFBLK   Block special.
  S_IFCHR   Character special.
  S_IFIFO   FIFO special.
  S_IFREG   Regular.
  S_IFDIR   Directory.
  S_IFLNK   Symbolic link.
  S_IFSOCK  Socket.
```

For reference, `fd` offers the following filetype filters currently, according to `--help`:
```
    -t, --type <filetype>...
            Filter the search by type (multiple allowable filetypes can be specified):
              'f' or 'file':         regular files
              'd' or 'directory':    directories
              'l' or 'symlink':      symbolic links
              'x' or 'executable':   executables
              'e' or 'empty':        empty files or directories
```

**Comment by sharkdp** [comment]
Thank you for the feedback. The problem is that sockets, FIFOs, etc. are specific to Unix-like operating systems, whereas `fd` tries to be platform independent.

Sure, we could OS-specific features, but that would complicate a lot of things (the code base, documentation, testing).

**Comment by tmccombs** [comment]
I'm not sure if it would meet @mcint's needs, but what if there was a "special" type to match files that are not "normal", which would include sockets, fifos, character/block special files, etc. on unix, and any similar os-specific specail files on windows if there are any.

**Comment by sharkdp** [comment]
The question is if this is really a 80% use case (or even a 99% use case) or if this is just a request to add these filetypes "for completeness".

So far, I've never found myself in the need to search for sockets or FIFOs - but obviously that doesn't mean that we should definitely not add this to `fd`.

**Comment by qubidt** [comment]
> The question is if this is really a 80% use case (or even a 99% use case) or if this is just a request to add these filetypes "for completeness".

Not saying this to be demanding, but I personally needed this today. I was `tar`ing some stuff and realized `tar format cannot archive socket` (understandably). I wanted to figure out which files were causing that and assumed I could depend on `fd` (which, other than this issue, has been a complete godsend).

Of course I can't speak to how much usage this would actually get, but I would say that this is a non-existing feature I find harder to work around than other (in comparison to `find`) missing features.

> Thank you for the feedback. The problem is that sockets, FIFOs, etc. are specific to Unix-like operating systems, whereas fd tries to be platform independent.

I definitely understand that tho (in the general, I personally value Unix-portability more than Windows-compatibility). I respect your decision either way.

**Comment by sharkdp** [comment]
Fixed via #572

**Comment by sharkdp** [comment]
This has now been released in [fd v8.0](https://github.com/sharkdp/fd/releases/tag/v8.0.0).
